docs/fastdds/dds_layer/domain/domainParticipant/domainParticipant.rst

```diff
@@ -1,5 +1,167 @@
+.. _dds_layer_domainParticipant:
 
 DomainParticipant
 =================
 
+A DomainParticipant is the entry point of the application to a domain.
+Every :ref:`dds_layer_domainParticipant` is linked to a single domain from its creation,
+and contains all the Entities related to that domain.
+It also acts as a factory for Publishers, Subscribers and Topics on that domain.
+
+The behavior of the DomainParticipant can be modified with the QoS values
+specified on :ref:`dds_layer_domainParticipantQos`.
+The QoS values can be set at the creation of the DomainParticipant,
+or modified later with :func:`set_qos()` method.
+
+Like other Entities, DomainParticipant accepts a Listener that will be notified of
+status changes on the DomainParticipant instance.
+
+
+.. _dds_layer_domainParticipant_creation:
+
+Creating a DomainParticipant
+----------------------------
+
+.. _dds_layer_domainParticipant_creation_qos:
+
+QoS based creation of a DomainParticipant
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Creation of a DomainParticipant is done with the :func:`create_participant()` method on the
+:ref:`dds_layer_domainParticipantFactory` singleton.
+
+The only mandatory parameter is the DomainId that identifies the domain where the DomainParticipant
+will be created.
+
+Optional parameters include:
+
+ * :ref:`dds_layer_domainParticipantQos` describing the behavior of the DomainParticipant.
+   If no value is provided, or if the provided value is :class:`PARTICIPANT_QOS_DEFAULT`,
+   the value of the :ref:`dds_layer_defaultDomainParticipantQos` is used.
+
+ * A Listener derived from :ref:`dds_layer_domainParticipantListener`, implementing the callbacks
+   that will be triggered in response to events and state changes on the DomainParticipant.
+   By default empty callbacks are used.
+
+ * A :class:`StatusMask` that activates or deactivates triggering of individual callbacks on the Listener.
+   By default all events are enabled.
+
+:func:`create_participant()` will return a null pointer if there was an error during the operation.
+For example, if the provided QoS is not compatible or is not supported.
+It is advisable to check that the returned value is a valid pointer.
+
+.. literalinclude:: /../code/DDSCodeTester.cpp
+   :language: c++
+   :start-after: //DDS_CREATE_DOMAINPARTICIPANT
+   :end-before: //!
+
+
+.. _dds_layer_domainParticipant_creation_profile:
+
+Profile based creation of a DomainParticipant
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Creation of a DomainParticipant based on a profile is done with the :func:`create_participant_with_profile()`
+method on the :ref:`dds_layer_domainParticipantFactory` singleton.
+
+This method takes two mandatory parameters:
+
+ * The DomainId that identifies the domain where the DomainParticipant will be created.
+
+ * The name of the profile to be applied to the DomainParticipant.
+
+Optional parameters include a Listener and a StatusMask (see :ref:`dds_layer_domainParticipant_creation_qos`).
```

Comment:
In general, I'd refer to the listener as a :class:`DomainParticipantListener` just to be extra specific. Actually, as a general comment to the Domain module section, I think that all reference to classes should use the :class: directive (attr for attributes, func for functions and so on).

Reply:
I made that paragraph clearer.

Reply:
Added the class and func directives.